File: app/src/main/java/com/antony/muzei/pixiv/provider/ClearCacheWorker.java
/*
 *     This file is part of PixivforMuzei3.
 *
 *     PixivforMuzei3 is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program  is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.antony.muzei.pixiv.provider;

import android.content.Context;
import android.os.Environment;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.io.File;

public class ClearCacheWorker extends Worker
{
    public ClearCacheWorker(
            @NonNull Context context,
            @NonNull WorkerParameters params)
    {
        super(context, params);
    }

    @NonNull
    @Override
    public Result doWork()
    {
        File dir = getApplicationContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
        String[] children = dir.list();
        for (String child : children)
        {
            new File(dir, child).delete();
        }
        PixivArtWorker.enqueueLoad(true, getApplicationContext());
        return Result.success();
    }
}

File: app/src/main/java/com/antony/muzei/pixiv/provider/ClearCacheWorker.kt
/*
 *     This file is part of PixivforMuzei3.
 *
 *     PixivforMuzei3 is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program  is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.antony.muzei.pixiv.provider

import android.content.Context
import android.os.Environment
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.antony.muzei.pixiv.provider.PixivArtWorker.Companion.enqueueLoad
import java.io.File

class ClearCacheWorker(
        context: Context,
        params: WorkerParameters) : Worker(context, params) {
    override fun doWork(): Result {
        val dir = applicationContext.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
        for (child in dir!!.list()) {
            File(dir, child).delete()
        }
        enqueueLoad(true, applicationContext)
        return Result.success()
    }
}
